src/app-service-registration.ts

```diff
@@ -196,7 +226,12 @@ export class AppServiceRegistration {
             regex: regex
         };
 
-        this.namespaces[type].push(regexObject);
+        const namespace = this.namespaces[type];
```

Comment:
It looks like `this.namespaces` is defined as optional. Should we protect from `this.namespaces` being `undefined`?

The usage of `addRegexPattern` is already behind some checks for `this.namespaces` so we're probably good for the current cases.

But maybe we want to set `this.namespaces = {};` if it doesn't exist already

Reply:
Nevermind, I see this refers to

```
private namespaces: {
        users: RegexObj[];	        users?: RegexObj[];
        aliases: RegexObj[];	        aliases?: RegexObj[];
        rooms: RegexObj[];	        rooms?: RegexObj[];
    } = { users: [], aliases: [], rooms: []};	    } = {};
```